frictionless/plugins/ckan.py

```diff
@@ -230,32 +223,61 @@ def write_table_convert_field_type(self, type):
             "object": "json",
             "array": "text[]",
             "geojson": "json",
-            "date": "date",
+            "date": "text",
             "time": "time",
             "year": "int",
             "datetime": "timestamp",
         }
-        try:
-            return DESCRIPTOR_TYPE_MAPPING[type]
-        except KeyError:
-            log.warn("Unsupported descriptor type '{}'.".format(type))
-            return None
+        return DESCRIPTOR_TYPE_MAPPING.get(type, "text")
 
-    def write_table_row_stream(self, name, row_stream, method="upsert"):
-        table = self.read_table(name)
+    def _write_table_row_stream(self, name, row_stream):
+        table = self._read_table(name)
         datastore_upsert_url = "{}/datastore_upsert".format(self.__base_endpoint)
-        records = []
-        for r in row_stream:
-            #  r = self.__mapper.convert_row(r, table.eschema)
-            records.append(r)
+        records = [r.to_dict(json=True) for r in row_stream]
```

Comment:
TODO: We're basically jamming the whole dataset into a single JSON payload here. From a practical perspective any real-life CKAN instance will have a limit on the maximum size of an HTTP POST body that it will accept so for large datasets this operation is going to fail.
This is copied from the https://github.com/frictionlessdata/datapackage-pipelines-ckan-driver implementation so it doesn't represent a regression, but as a future piece of work we should really give the option to split the dataset into chunks here for dealing with large datasets. Suggest logging under #475